File: day01/ruby/anoff/solution_test.rb
require "test/unit"
require_relative './solution'

class TestSolution < Test::Unit::TestCase
  def test_required_fuel
    d = {
      12 => 2,
      14 => 2,
      1969 => 654,
      100756 => 33583
    }
    d.each{ |mass, fuel|
      assert_equal fuel, required_fuel(mass), "Incorrect fuel for mass=%d" % mass
    }
  end
end

File: day01/go/anoff/solution_test.go
package main

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestRequiredFuel(t *testing.T) {
	m := make(map[int]int)
	m[12] = 2
	m[14] = 2
	m[1969] = 654
	m[100756] = 33583
	for mass, fuel := range m {
		res := requiredFuel(mass)
		assert.Equal(t, fuel, res, "Wrong result for mass=%d", mass)
	}
}

func TestRequiredFuelRecursive(t *testing.T) {
	m := make(map[int]int)
	m[12] = 2
	m[14] = 2
	m[1969] = 966
	m[100756] = 50346
	for mass, fuel := range m {
		res := requiredFuelRecursive(mass)
		assert.Equal(t, fuel, res, "Wrong result for mass=%d", mass)
	}
}
